tensorflow/compiler/tf2tensorrt/convert/ops/resize.cc

```diff
@@ -0,0 +1,288 @@
+    return {DataType::DT_FLOAT, DataType::DT_HALF, DataType::DT_INT32};
+  }
+
+  Status Validate() {
+    const auto& inputs = params_->inputs;
+    const auto& node_def = params_->node_def;
+
+    // Get input tensor.
+    input_tensor_ = inputs.at(0).tensor();
+    TRT_ENSURE(input_tensor_ != nullptr);
+
+    // Check whether spatial dim is dynamic.
+    StatusOr<bool> static_spatial_input_shape =
+        CHWDimsStaticSpatialShape(input_tensor_->getDimensions());
+    TRT_ENSURE_OK(static_spatial_input_shape);
+    attrs_.static_output_shape =
+        (*static_spatial_input_shape) && inputs.at(1).is_weights();
+
+    // Check output size. It must constain two values i.e. [H_out, W_out]
+    if (inputs.at(1).is_weights()) {
+      // Output size is given as a constant.
+      if (inputs.at(1).weights().count() != 2) {
+        return errors::Unimplemented(
+            "Resize requires 2D values for the size, at ", node_def.name());
+      }
+    } else {
+      // Output size is given as a tensor, possibly as the result of shape
+      // calculation ops in the graph.
+      if (params_->use_implicit_batch) {
+        return errors::Unimplemented(
+            "Resize requires constant size in implicit batch mode, at ",
+            node_def.name());
+      }
+
+      // Check that this is a shape tensor and it has two values.
+      ITensorProxyPtr size_param = inputs.at(1).tensor();
+      TRT_ENSURE(size_param->getType() == nvinfer1::DataType::kINT32);
+      TRT_ENSURE(size_param->getDimensions().nbDims == 1);
+      if (size_param->getDimensions().d[0] != 2) {
+        return errors::Unimplemented(
+            "Resize requires 2D values for the size, at ", node_def.name());
+      }
+    }
+
+    // Verify and consume node attributes.
+    StatusOr<bool> align_corners = GetAttrValue<bool>("align_corners");
+    TRT_ENSURE_OK(align_corners);
+    StatusOr<bool> half_pixel_centers =
+        GetAttrValue<bool>("half_pixel_centers");
+    TRT_ENSURE_OK(half_pixel_centers);
+    attrs_.half_pixel_centers = *half_pixel_centers;
+    attrs_.align_corners = *align_corners;
+
+    if (node_def.op() == "ResizeBilinear") {
+      attrs_.mode = nvinfer1::ResizeMode::kLINEAR;
+    } else if (node_def.op() == "ResizeNearestNeighbor") {
+      attrs_.mode = nvinfer1::ResizeMode::kNEAREST;
+    } else {
+      return errors::Unimplemented(node_def.op(), " is not yet implemented at ",
+                                   node_def.name());
+    }
+
+    // If we require workaround and have dynamic spatial shape, this is not
+    // supported.
+    if (!attrs_.static_output_shape && !attrs_.align_corners &&
+        attrs_.mode == nvinfer1::ResizeMode::kLINEAR) {
+      return errors::Unimplemented(
+          "Unsupported (bilinear, align corners=False, dynamic spatial shape)");
+    }
+    return Status::OK();
+  }
+
+  Status Convert() {
+    // Transpose tensor from NHWC to NCHW format.
+    // TODO(cbate): Check if reformat is required.
```

Comment:
Do you mean TensorRT may not require NCHW?
Can you create a github ticket to follow up on this and put the ticket number in the TODO?

Reply:
This was incorrect comment. Bilinear resize op always is NHWC in TF so we need to insert the symbolic transpose for TRT.